scylla-cql/src/types/serialize/value.rs

```diff
@@ -2069,6 +2069,44 @@ mod tests {
         assert_eq!(reference, udt);
     }
 
+    #[test]
+    fn test_udt_serialization_with_missing_rust_field() {
+        let udt = TestUdtWithFieldSorting::default();
+
+        let typ_normal = ColumnType::UserDefinedType {
+            type_name: "typ".to_string(),
+            keyspace: "ks".to_string(),
+            field_types: vec![
+                ("a".to_string(), ColumnType::Text),
+                ("b".to_string(), ColumnType::Int),
+                (
+                    "c".to_string(),
+                    ColumnType::List(Box::new(ColumnType::BigInt)),
+                ),
+            ],
+        };
+
+        let typ_unexpected_field = ColumnType::UserDefinedType {
+            type_name: "typ".to_string(),
+            keyspace: "ks".to_string(),
+            field_types: vec![
+                ("a".to_string(), ColumnType::Text),
+                ("b".to_string(), ColumnType::Int),
+                (
+                    "c".to_string(),
+                    ColumnType::List(Box::new(ColumnType::BigInt)),
+                ),
+                // Unexpected field
+                ("d".to_string(), ColumnType::Counter),
+            ],
+        };
+
+        let result_normal = do_serialize(&udt, &typ_normal);
+        let result_additional_field = do_serialize(&udt, &typ_unexpected_field);
+
+        assert_eq!(result_normal, result_additional_field);
+    }
```

Comment:
In addition to those tests, I'd like to see the new behavior tested in `cql_types_test.rs` with a real Scylla cluster, to see whether sending one field less does actually work.